crates/node/src/lib.rs

```diff
@@ -654,6 +661,73 @@ impl Node {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ProposalManager {
+    ctx_manager: Arc<ContextManager>,
+}
+
+impl ProposalManager {
+    pub fn new(ctx_manager: ContextManager) -> Self {
+        Self {
+            ctx_manager: Arc::new(ctx_manager),
+        }
+    }
+
+    pub async fn get_number_of_proposal_approvals(
+        &self,
+        context_id: &ContextId,
+        proposal_id: &ProposalId,
+    ) -> EyreResult<ProposalWithApprovals> {
+        self.ctx_manager
+            .get_number_of_proposal_approvals(*context_id, *proposal_id)
+            .await
+    }
+
+    pub async fn get_proposal_approvers(
+        &self,
+        context_id: &ContextId,
+        proposal_id: &ProposalId,
+    ) -> EyreResult<Vec<ContextIdentity>> {
+        self.ctx_manager
+            .get_proposal_approvers(*context_id, *proposal_id)
+            .await
+    }
+
+    pub async fn get_active_proposals_count(&self, context_id: &ContextId) -> EyreResult<u16> {
+        self.ctx_manager
+            .get_number_of_active_proposals(*context_id)
+            .await
+    }
+
+    pub async fn get_proposals(
+        &self,
+        context_id: &ContextId,
+        offset: usize,
+        limit: usize,
+    ) -> EyreResult<Vec<Proposal>> {
+        self.ctx_manager
+            .get_proposals(*context_id, offset, limit)
+            .await
+    }
+
+    pub async fn get_proposal(
+        &self,
+        context_id: &ContextId,
+        proposal_id: &ProposalId,
+    ) -> EyreResult<Option<Proposal>> {
+        let result = self
+            .ctx_manager
+            .get_proposal(*context_id, *proposal_id)
+            .await;
+
+        match result {
+            Ok(proposal) => Ok(Some(proposal)),
+            Err(err) if err.to_string().contains("no proposal found") => Ok(None),
+            Err(err) => Err(err),
+        }
+    }
+}
```

Comment:
so `ProposalManager` is a wrapper of `ContextManager` that adds no extra functionality?

Reply:
btw, #1278 merged which heavily refactored large parts of the codebase, which includes context manager, I'd say this covers the thinking that might've led to your `ProposalManager`